examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -195,10 +201,40 @@ static void wfx_rsi_join_cb(uint16_t status, const uint8_t * buf, const uint16_t
          * We should enable retry.. (Need config variable for this)
          */
         WFX_RSI_LOG("%s: failed. retry: %d", __func__, wfx_rsi.join_retries);
-#if (WFX_RSI_CONFIG_MAX_JOIN != 0)
-        if (++wfx_rsi.join_retries < WFX_RSI_CONFIG_MAX_JOIN)
-#endif
+        if (!is_disconnection_event)
+        {
+            /* After the reboot or a commissioning time device failed to connect with AP.
+             * Device will retry to connect with AP upto WFX_RSI_CONFIG_MAX_JOIN retries.
+             */
+            if (wfx_rsi.join_retries < WFX_RSI_CONFIG_MAX_JOIN)
+            {
+                WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(WLAN_RETRY_TIMER_MS));
+                vTaskDelay(pdMS_TO_TICKS(WLAN_RETRY_TIMER_MS));
+                xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+            }
+            else
+            {
+                WFX_RSI_LOG("Connect failed after %d tries", wfx_rsi.join_retries);
+            }
+        }
+        else
         {
+            /* After disconnection
+             * At the telescopic time interval device try to reconnect with AP, upto WLAN_MAX_RETRY_TIMER_MS intervals
+             * are telescopic. If interval exceed WLAN_MAX_RETRY_TIMER_MS then it will try to reconnect at
+             * WLAN_MAX_RETRY_TIMER_MS intervals.
+             */
+            if (retryInterval > WLAN_MAX_RETRY_TIMER_MS)
+            {
+                retryInterval = WLAN_MAX_RETRY_TIMER_MS;
+                WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(WLAN_MAX_RETRY_TIMER_MS));
+            }
+            else
+            {
+                WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(retryInterval));
+            }
+            vTaskDelay(pdMS_TO_TICKS(retryInterval));
+            retryInterval += retryInterval;
```

Comment:
Pattern #1

Reply:
Pattern https://github.com/project-chip/connectedhomeip/pull/1 and https://github.com/project-chip/connectedhomeip/issues/2 are slight different whereas for pattern https://github.com/project-chip/connectedhomeip/pull/1 and https://github.com/project-chip/connectedhomeip/pull/3 are same but used in rs9116 and wf200 resp so only one file will get complied at a time.

In pattern https://github.com/project-chip/connectedhomeip/pull/1 we have an extra if condition to check retries before giving the delay whereas in pattern https://github.com/project-chip/connectedhomeip/issues/2 we are directly giving the delay without any condition.

Reply:
> Pattern #1 and #2 are slight different whereas for pattern #1 and #3 are same but used in rs9116 and wf200 resp so only one file will get complied at a time.
doesn't matter you can either create a common file or create a specific function for this (can be a callback w/ param)


> In pattern #1 we have an extra if condition to check retries before giving the delay whereas in pattern #2 we are directly giving the delay without any condition.